### PR TITLE
Add dydx validators gentx and pregenesis for testnet #4

### PR DESCRIPTION
**Pull request by teddyding**

### CODE REVIEW

File: validate.sh [hunk 1]
```diff
@@ -18,6 +18,11 @@ dydxprotocold_path=$(tar -xvf $tar_path --directory /tmp/dydxprotocold)
 mkdir -p ~/bin
 cp /tmp/dydxprotocold/$dydxprotocold_path /usr/local/bin/dydxprotocold
 
+NINE_ZEROS="000000000"
+EIGHTEEN_ZEROS="$NINE_ZEROS"
+INITIAL_VALIDATOR_TOKEN_BALANCE="1000000$EIGHTEEN_ZEROS"
+INITIAL_VALIDATOR_SELF_DELEGATION="500000$EIGHTEEN_ZEROS"
```
Comment (teddyding) [inline validate.sh:24]: This requires `v0.4.0-rc0` binary to be accepted by the validation workflow @2dpetkov

File: validate.sh [hunk 2]
```diff
@@ -18,6 +18,11 @@ dydxprotocold_path=$(tar -xvf $tar_path --directory /tmp/dydxprotocold)
 mkdir -p ~/bin
 cp /tmp/dydxprotocold/$dydxprotocold_path /usr/local/bin/dydxprotocold
 
+NINE_ZEROS="000000000"
+EIGHTEEN_ZEROS="$NINE_ZEROS"
```
Comment (ttl33) [inline validate.sh:22]: i think there should be another nine zeros

```suggestion
EIGHTEEN_ZEROS="$NINE_ZEROS$NINE_ZEROS"
```

File: dydx-testnet-4/pregenesis.json [hunk 3]
```diff
@@ -0,0 +1,2020 @@
+{
+  "genesis_time": "2023-10-11T04:19:58.475018Z",
```
Comment (ttl33) [inline dydx-testnet-4/pregenesis.json:2]: I think this is just a dummy pregenesis right?

otherwise, we should update this to be 10/17 17:00 UTC

**Review by ttl33 — APPROVED** [review]

File: dydx-testnet-4/pregenesis.json (hunk 3 above)
Comment (teddyding) [reply dydx-testnet-4/pregenesis.json:2]: Yes

File: validate.sh (hunk 2 above)
Comment (teddyding) [reply validate.sh]: Done!

**Pull request merged**